rest/api.go

```diff
@@ -0,0 +1,80 @@
+package rest
+
+import (
+	"github.com/Peripli/service-manager/pkg/filter"
+	"github.com/Peripli/service-manager/pkg/plugin"
+	"github.com/sirupsen/logrus"
+)
+
+// API is the primary point for REST API registration
+type API struct {
+	// Controllers contains the registered controllers
+	Controllers []Controller
+
+	// Filters contains the registered filters
+	Filters []filter.Filter
+}
+
+// RegisterControllers registers a set of controllers
+func (api *API) RegisterControllers(controllers ...Controller) {
+	for _, controller := range controllers {
+		if controller == nil {
+			logrus.Panicln("Cannot add nil controllers")
+		}
+		api.Controllers = append(api.Controllers, controller)
+	}
+}
+
+// RegisterFilters registers a set of filters
+func (api *API) RegisterFilters(filters ...filter.Filter) {
+	api.Filters = append(api.Filters, filters...)
+}
+
+// RegisterPlugins registers a set of plugins
+func (api *API) RegisterPlugins(plugins ...plugin.Plugin) {
+	for _, plug := range plugins {
+		if plug == nil {
+			logrus.Panicln("Cannot add nil plugins")
+		}
+		match := false
+		register := func(method string, pathPattern string, middleware filter.Middleware) {
```

Comment:
extract register func to return always true and set match = register(...) everywhere